src/lib/webhookserver.js

```diff
@@ -0,0 +1,147 @@
+import {spawn} from 'child_process';
+import Debug from 'debug';
+import Promise from 'promise';
+import http from 'http'
+import path from 'path';
+import waitForPort from '../wait_for_port';
+
+let debug = Debug('docker-worker-runtime:webhookserver');
+
+const DEFAULT_TARGET_PORT = 61022;
+
+class WebhookServer {
+  /**
+   * Create a new instance of webhookserver.
+   * The whclient binary must be in bin-utils.
+   * Default target port is 61022.
+   *
+   * @param {String} binPath - Location of whclient binary
+   * @param {String} clientID - Taskcluster client ID
+   * @param {String} accessToken - Taskcluster access token
+   * @param {String} targetPort - Port to which connections should
+   * forwared
+   */
+  constructor({clientId, accessToken}, targetPort){
```

Comment:
temporary credentials include a `certificate`, just pass creds around as an object and you'll be fine...